### PR TITLE
find_maximum_value_from_a_list_of_numerical_input/data

### PR DESCRIPTION
**Pull request by Allukmon**
---
name: solution review 
about: A template PR for code review with a checklist
---

# Behavior

## Files

- [ ] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [ ] The function's behavior is described
- The function's arguments are described:
  - [ ] Type
  - [ ] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [ ] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [ ] The function's name describes it's behavior
- [ ] The function's name matches the file name
- [ ] The function has correct type annotations

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

**Comment by Awaab98** [comment]
Hello @Allukmon, I will take on the review process for this implementation.

**Comment by Allukmon** [comment]
Thanks a lot

**Comment by Allukmon** [comment]
Thanks a lot Awaab > Hello @Allukmon, I will take on the review process for this implementation.

**Comment by Awaab98** [comment]
I have started the review process. There are **two issues** that you should address. Firstly, I checked the **files changed** section in the pull request. You have only added the function python file, and sill remains the test file. Additionally, the function file is empty and you need to push your latest locall implementation to your local branch and then update the pull request. Secondly, after you finished writing the function and the test files, run **ruff format path/to/file.py** command on your CLI to pass the 2 remaining the CI checks related to py_formating.

**Comment by Allukmon** [comment]
@Awaab98  thanks for your help, I have been able to pass all the CI-checks

**Comment by Awaab98** [comment]
Ok. Now that you have passed the CI checks, lets start reviewing your test and function python files.

**Comment by Allukmon** [comment]
please go ahead with the review

**Comment by Allukmon** [comment]
@Awaab98 you might want to show me where you made the suggestion because I can not find it on the file I am working on

**Review by Awaab98 — CHANGES_REQUESTED** [review]
Please review the suggested changes.

File: solutions/find_maximum_value.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        4
+        >>> find_maximum_value([2.24, 4, 5.23, 1])
+        5.23
+        >>> find_maximum_value([1.2, 3.4, 2.7, 0.5])
+        3.4
+    """
+
+    # trunk-ignore(bandit/B101)
+    assert isinstance(values, list)
+    # trunk-ignore(bandit/B101)
```
[the remark this answers is not in the document]
Comment (Allukmon) [reply solutions/find_maximum_value.py:39]: It is to ignore the error on the line; I have removed it

File: solutions/find_maximum_value.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    takes a list of numbers and return a single number which is the highest value from the list.
+
+    Parameters:
+        values: list, the input values to process
+
+    Returns -> number: the highest value from the list of numbers.
+
+    Raises:
+        AssertionError: if input is not a list or a non-numerical data
```
[the remark this answers is not in the document]
Comment (Allukmon) [reply solutions/find_maximum_value.py:26]: Thanks for pointing that out

**Review by Awaab98 — APPROVED** [review]
